### DISCUSSION TITLE
Suggestion on AtomFamily liked feature

### DISCUSSION THREAD

**Discussion by tpatalas** [pmndrs/zustand#1000] [Q&A] [answered]
I am currently in transition from Recoil to Zustand on my project. 

The Recoil is excellent, but the layer of the complexities and difficulties in maintenances exponentially grows as the project becomes larger. Zustand...is on the other hand, just light-weighted, but it just works. No wonder Zustand is so popular. 

My current challenge in Zustand is the SHARABLE local state. I was achieving this with `atomFamily` in Recoil. (I believe Jotai has one too). This was quite useful as we isolate each state with given ID without degrading performance. 

For Example:
```ts
// such as in Recoil
// initial atomFamilyTodo value is `false`
 const todo = useRecoilValue(atomFamilyTodo(todoID))

// This creates:
// ...
// todo with ID 1 false
// todo with ID 2 false
// todo with ID 3 false
// ...

// state is isolated with the given ID
```

As my knowledge about Zustand is not so deep yet, the only solutions I can think of are

1. Just use `useState` as a local state without sharing the state (or Maybe with context or prop drills)
2. Create Store within the reusable component [#101](https://github.com/pmndrs/zustand/issues/101)

Is there an option or suggestion to achieve the Sharable local state in Zustand?

**Comment by dai-shi** [comment, marked as answer]
This requires some understanding of React rules, but something like this should be possible.
```js
const todoStoreCache = new Map()
const storeFamilyTodo = (todoId) => {
  let store = todoStoreCache.get(todoId)
  if (!store) {
    store = create(...)
    todoStoreCache.set(todoId, store)
  }
  return store
}

const Component = () => {
  const useStore = storeFamilyTodo('foo');
  // ...
}
```

You could even create a `storeFamily` util function.

However, we don't usually do it in zustand. We often create a single store instead:
```js
const useStoreTodos = create((set) => ({
  todos: [],
  // ...
}))
```

**Reply by tpatalas** [reply]
Great! Thank you for the tip although It maybe better to followup the general rule of the Zustand, creating single store instead of creating many.

I ended up refactoring the architecture instead of  applying same rules that I had before. Thank you for the tip again and really appreciate for the GREAT library!
